TestCombine.playground/Pages/CreatePublisher.xcplaygroundpage/Contents.swift

```diff
@@ -118,3 +118,105 @@ c1?.cancel()
 //}
 //
 //print(c)
+
+public class Observer<Input, Failure: Error> {
+
+    @Synchronized private var isCancelled: Bool = false
+    private let receiveClosure: (Input) -> Subscribers.Demand
+    private let receiveCompletionClosure: (Subscribers.Completion<Failure>) -> Void
+
+    init(receive: @escaping (Input) -> Subscribers.Demand, receiveCompletion: @escaping (Subscribers.Completion<Failure>) -> Void) {
+        receiveClosure = receive
+        receiveCompletionClosure = receiveCompletion
+    }
+
+    public func receive(_ input: Input) -> Subscribers.Demand {
+        guard isCancelled == false else { return .none }
+        return receiveClosure(input)
+    }
+
+    public func receive(completion: Subscribers.Completion<Failure>) -> Void {
+        guard isCancelled == false else { return }
+        return receiveCompletionClosure(completion)
+    }
+
+    fileprivate func cancel() {
+        isCancelled = true
+    }
+}
+
+public struct Orchestrator {
+    public let stop: () -> Void
+    public let start: (Subscribers.Demand) -> Void
+}
+
+public typealias CreationTask<Input, Failure: Error> = (Observer<Input, Failure>, Subscribers.Demand) -> Orchestrator
+
+public struct Create<Output, Failure: Error>: Publisher {
+
+    private let task: CreationTask<Output, Failure>
+    private let demand: Subscribers.Demand
+
+    public init(maxValues: Subscribers.Demand = .unlimited, task: @escaping CreationTask<Output, Failure>) {
+        self.demand = maxValues
+        self.task = task
+    }
+
+    public func receive<S>(subscriber: S) where S : Subscriber, Failure == S.Failure, Output == S.Input {
+        subscriber.receive(subscription: CreateSubscription(publisher: self, subscriber: AnySubscriber(subscriber), maxValues: demand))
+    }
+
+    private class CreateSubscription<Input, Failure: Error>: Subscription {
+
+        @Synchronized private var subscriber: AnySubscriber<Input, Failure>?
+        private var orchestrator: Orchestrator?
+
+        init(publisher: Create<Input, Failure>, subscriber: AnySubscriber<Input, Failure>, maxValues: Subscribers.Demand) {
+            self.subscriber = subscriber
+
+            let observer = Observer<Input, Failure>(
+                receive: { [weak self] input in
+                    self?.subscriber?.receive(input) ?? .none
+                },
+                receiveCompletion: { [weak self] completion in
+                    self?.subscriber?.receive(completion: completion)
+            })
+
+            orchestrator = publisher.task(observer, maxValues)
+        }
+
+        func request(_ demand: Subscribers.Demand) {
+            orchestrator?.start(demand)
+        }
+
+        func cancel() {
+            subscriber = nil
+            orchestrator?.stop()
+        }
+    }
+}
+
+let c = Create<Int, Never>(maxValues: .max(10)) { observer, maxValues in
```

Comment:
Each client of this api must handle manually the demand!
Not a very easy to use api...

Reply:
Will remove because useless, maxValues here will then be .max(10) ... so really overkill
👍